### ISSUE TITLE
BUG : `GetVsDbg.ps1` does not handle proxy

### ISSUE THREAD

**Issue by tebeco** [Microsoft/DockerTools#38]
```
function DownloadAndExtract([string]$url, [string]$targetLocation) {
    Add-Type -assembly "System.IO.Compression.FileSystem"
    Add-Type -assembly "System.IO.Compression"
    $zipStream = (New-Object System.Net.WebClient).OpenRead($url)
    $zipArchive = New-Object System.IO.Compression.ZipArchive -ArgumentList $zipStream
    [System.IO.Compression.ZipFileExtensions]::ExtractToDirectory($zipArchive, $targetLocation)
    $zipArchive.Dispose()
    $zipStream.Dispose()
}
```

This is really bad for handle Proxy well it does not care, and there is no workaround
I've almost never done any powershell, but from what i found here
https://stackoverflow.com/questions/571429/powershell-web-requests-and-proxies

here is a working way to at least read default proxy settings
I did not test this change on a computer without a proxy BTW
But at least it helps working in Corp Env.

```
function DownloadAndExtract([string]$url, [string]$targetLocation) {
    Add-Type -assembly "System.IO.Compression.FileSystem"
    Add-Type -assembly "System.IO.Compression"

    $proxy = [System.Net.WebRequest]::GetSystemWebProxy()
    $proxy.Credentials = [System.Net.CredentialCache]::DefaultCredentials
    $wc = new-object system.net.WebClient
    $wc.proxy = $proxy

    # $zipStream = (New-Object System.Net.WebClient).OpenRead($url)
    $zipStream = $wc.OpenRead($url)
    $zipArchive = New-Object System.IO.Compression.ZipArchive -ArgumentList $zipStream
    [System.IO.Compression.ZipFileExtensions]::ExtractToDirectory($zipArchive, $targetLocation)
    $zipArchive.Dispose()
    $zipStream.Dispose()
}

```

**Comment by rajkumar42** [comment]
@tebeco did you have to configure your browser IE/Edge/Chrome to work? Can you also check if Control Panel -> Internet Options -> Connections tab -> LAN Settings -> Proxy Server is set?

**Comment by tebeco** [comment]
@rajkumar42
just to be sure,

the code i added, READ the system proxy and it worked after that
So it means, yes my proxy was setup before i gave a shot
and the GetVsDvg.ps1 file try to do http without using / reading system proxy....

so my proxy is set (by a GPO in my company) every software works well unless the one that does not try to read that proxy
like this ps1 file

**Comment by rajkumar42** [comment]
@tebeco this has been fixed and will available in the future version of Visual Studio. Thanks!

**Comment by tebeco** [comment]
Do you think going OpenSource would help développer to understand Docker especially case like this ?
Why is blocking the team for open sourcing it ?

**Comment by bwateratmsft** [comment]
This issue is being closed as an inactive issue. If you wish to keep it active, please re-open with any additional context for our consideration, we are happy to take a look!

**Comment by alandball** [comment]
> @tebeco this has been fixed and will available in the future version of Visual Studio. Thanks!

@rajkumar42 Hi, I'm still finding this to be an issue with the latest update of VS2017(15.9.5). 

To fix this, I separated the  DownloadAndExtract function in the script to Download(), and Extract(). '-s    Skips any steps which requires downloading from the internet.' This does not indicate that it will also skip the install. I have added an additional parameter for skip install.
